#### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:

-  на факультете A;  
-  на факультете B;  
-  на факультете C.

Это событие может произойти следующими способами:  
$P(B|A_1)$: студент поступил на факультет $A=n$ И $P(A_1)$ вероятность сдачи первой сессии = 0.8;  
$P(B|A_2)$: студент поступил на факультет $B=n$ И $P(A_2)$ вероятность сдачи первой сессии = 0.7;  
$P(B|A_3)$: студент поступил на факультет $C=2n$ И $P(A_3)$ вероятность сдачи первой сессии = 0.9;  

События $P(B|A_1)$, $P(B|A_2)$, и $P(B|A_3)$ образуют полную вероятность сдачи первой сессии = 1, значит $P(B|A_1)=\frac{1}{n}=0.25$, $P(B|A_2)=\frac{1}{n}=0.25$, $P(B|A_3)=\frac{1}{2n}=0.5$  
  
Вероятность того, что студент учится, определяется, как сумма вероятностей этих событий: $P(A)=P(B|A_1)*P(A_1)+P(B|A_2)*P(A_2)+P(B|A_3)*P(A_3)=\ 0.25\cdot 0.8+\ 0.25\cdot 0.7+\ 0.5\cdot 0.9$ 

In [6]:
PB = 0.25 * 0.8 + 0.25 * 0.7 + 0.5 * 0.9
print(f'Полная вероятность наступления события PA = {round(PB, 4)}.')

Полная вероятность наступления события PA = 0.825.


С помощью формулы Байеса $P(A|B)=\frac{P(B|A)\cdot P(A)}{P(B)}$ оценим вероятность того, на каком факультете учится студент, успешно сдавший сессию. :  
$P(A|B_1)=\frac{\ 0.25\cdot 0.8}{0.825}$  
$P(A|B_2)=\frac{\ 0.25\cdot 0.7}{0.825}$  
$P(A|B_3)=\frac{\ 0.5\cdot 0.9}{0.825}$  

In [8]:
PAB1 = 0.25 * 0.8 / PB
PAB2 = 0.25 * 0.7 / PB
PAB3 = 0.5 * 0.9 / PB
print(f'Вероятность того, что студент учится на факультете А: {round(PAB1, 4)} = {round(PAB1 * 100, 2)}%;\n'
      f'Вероятность того, что студент учится на факультете B: {round(PAB1, 4)} = {round(PAB1 * 100, 2)}%;\n'
      f'Вероятность того, что студент учится на факультете C: {round(PAB3, 4)} = {round(PAB3 * 100, 2)}%.')

Вероятность того, что студент учится на факультете А: 0.2424 = 24.24%;
Вероятность того, что студент учится на факультете B: 0.2424 = 24.24%;
Вероятность того, что студент учится на факультете C: 0.5455 = 54.55%.


### Результат:  

а) Вероятность того, что студент учится на факультете А: 24.24%;  
б) Вероятность того, что студент учится на факультете B: 24.24%;  
в) Вероятность того, что студент учится на факультете C: 54.55%.   